aeron-client/src/main/cpp_wrapper/util/Exceptions.h

```diff
@@ -138,7 +138,7 @@ AERON_DECLARE_SOURCED_EXCEPTION(UnknownSubscriptionException, ExceptionCategory:
 AERON_DECLARE_SOURCED_EXCEPTION(ReentrantException, ExceptionCategory::EXCEPTION_CATEGORY_ERROR);
 AERON_DECLARE_SOURCED_EXCEPTION(UnsupportedOperationException, ExceptionCategory::EXCEPTION_CATEGORY_ERROR);
 
-#define AERON_MAP_TO_SOURCED_EXCEPTION_AND_THROW(code, message)                     \
+#define AERON_MAP_TO_SOURCED_EXCEPTION_AND_THROW_WITH_DEFAULT(code, message, defaultException)  \
```

Comment:
This was reworked a bit so that I can throw an ArchiveException as the default when these macros are leveraged from the archive cpp_wrapper project.